### ISSUE TITLE
Can CogVideoX-2B model fit the 40gb-A100 gpu during training?

### ISSUE THREAD

**Issue by LongVu219** [snap-research/ac3d#27]
Hi, thanks for the great work. I wonder can we fit the model into 40gb-A100 gpus after enable all the slicing/optimization technique, what is the training time of this setting.

**Comment by sherwinbahmani** [comment]
You should check in the original repo: https://github.com/zai-org/CogVideo
There are some additonal things in the readme you can try if you can encounter OOM. I don't have any experience since I used 80GB GPUs

**Comment by wz0919** [comment]
I think the issue comes from
https://github.com/snap-research/ac3d/blob/3c1e29e688f4a6d0f0ad41f1bf75d2eab709dac2/cogvideo_transformer.py#L46

With this condition, gradient checkpointing is only enabled when the model is in training mode. In the code, transformer isn’t set to training, so checkpointing is not actually activated.

To enable it, you can simply remove the self.training check so that checkpointing is applied whenever self.gradient_checkpointing is set and it can significantly reduce memory usage.

**Comment by sherwinbahmani** [comment]
Thanks, I fixed it in the repo!
